src/pipeline.jl

```diff
@@ -331,9 +331,10 @@ function _override_seriestype_check(plotattributes::AKW, st::Symbol)
     if !RecipesPipeline.is3d(st) && !(st in (:contour, :contour3d))
         z = plotattributes[:z]
         if !isa(z, Nothing) &&
-           (size(plotattributes[:x]) == size(plotattributes[:y]) == size(z))
+           (size(plotattributes[:x]) == size(plotattributes[:y]) == size(z)) &&
+	   st !== :mesh3d
```

Comment:
This shouldn't be necessary since it should already been handled by `!RecipesPipeline.is3d(st)`